pkg/remotemetrics/resources.go

```diff
@@ -108,12 +108,24 @@ func (m *resourceGetter) GetNodeNames(ctx context.Context) []string {
 	})
 	utilruntime.Must(err)
 
-	res := make([]string, len(nodes.Items))
+	res := make([]string, 0)
 	for i := range nodes.Items {
 		node := &nodes.Items[i]
-		res[i] = node.Name
+		if nodeIsReady(node) {
+			res = append(res, node.Name)
+		}
 	}
 
 	klog.V(2).Infof("Scraping nodes %+v", res)
 	return res
 }
+
+func nodeIsReady(node *corev1.Node) bool {
+	conditions := node.Status.Conditions
+	for i := range conditions {
+		if conditions[i].Type == corev1.NodeReady {
+			return conditions[i].Status == corev1.ConditionTrue
+		}
+	}
+	return false
+}
```

Comment:
This function already exists as liqo utility: https://github.com/liqotech/liqo/blob/3461899762cc889892133ea7cf5b914353a7d6fb/pkg/utils/nodeUtils.go#L24-L31